convert_seg_to_doc.py

```
@@ -0,0 +1,25 @@
+parser = argparse.ArgumentParser()
+parser.add_argument('--input', required=True, help='input file path')
+parser.add_argument('--output', required=True, help='output file path')
+parser.add_argument('--hits', required=True, help='output file path')
+
+args = parser.parse_args()
+
+with open(args.input, 'r') as f:
+    with open(args.output, 'w') as fout:
+        query_ref = "query_ref"
+        for line in f:
+            query_id, a, doc_seg_id, rank, score, run_tag = line.split(' ')
+            if query_id != query_ref:
+                top_1000 = []
+                rank = 1
+            if rank <= args.hits:
+                doc_seg = doc_seg_id.split('#') 
+                doc_id = doc_seg[0]
+                if doc_id not in top_1000:
```

Similar to `trec_eval`, I think it should raise an exception if there is a duplicated doc id

---

Segment id is doc id + serial id of the segment in this doc. It is possible to have a few duplicate doc ids if we are trying to rank the segments since segments may come from the same document. Maybe it's not really an exception?

---

I see, makes sense. Just a minor thing: instead of using a list for `top_1000`, use a set. This will make `if doc_id not in top_1000` much faster.